apps/dotcom/sync-worker/src/TLPostgresReplicator.ts

```diff
@@ -434,166 +570,78 @@ export class TLPostgresReplicator extends DurableObject<Environment> {
-						event,
-						row,
-					})
+				default: {
+					const _x: never = table
+					this.captureException(new Error(`Unhandled table: ${table}`), { change })
 					break
+				}
 			}
 		} catch (e) {
 			this.captureException(e)
 		}
 	}
 
-	private getRowAndEvent(change: Wal2Json.Change) {
-		const row = {} as any
-		// take everything from change.columnnames and associated the values from change.columnvalues
-		if (change.kind === 'delete') {
-			const oldkeys = change.oldkeys
-			assert(oldkeys, 'oldkeys is required for delete events')
-			assert(oldkeys.keyvalues, 'oldkeys is required for delete events')
-			oldkeys.keynames.forEach((key, i) => {
-				row[key] = oldkeys.keyvalues[i]
-			})
-		} else {
-			change.columnnames.forEach((col, i) => {
-				row[col] = change.columnvalues[i]
-			})
-		}
-
-		const event = {
-			command: change.kind,
-			table: change.table,
-		}
-		return { row, event }
-	}
-
-	private handleBootEvent(row: Row | null, event: ReplicationEvent) {
-		if (event.command === 'delete') return
-		assert(row && 'bootId' in row, 'bootId is required')
-		this.messageUser(row.userId, {
-			type: 'boot_complete',
-			userId: row.userId,
-			bootId: row.bootId,
-		})
-	}
-
-	private handleMutationConfirmationEvent(row: Row | null, event: ReplicationEvent) {
+	private handleMutationConfirmationEvent(
+		collator: UserChangeCollator,
+		row: Row | null,
+		event: ReplicationEvent
+	) {
 		if (event.command === 'delete') return
 		assert(row && 'mutationNumber' in row, 'mutationNumber is required')
-		this.messageUser(row.userId, {
+		collator.addChange(row.userId, {
 			type: 'mutation_commit',
 			mutationNumber: row.mutationNumber,
 			userId: row.userId,
 		})
 	}
 
-	private handleFileStateEvent(row: Row | null, event: ReplicationEvent) {
+	private handleFileStateEvent(
+		collator: UserChangeCollator,
+		row: Row | null,
+		event: ReplicationEvent
+	) {
 		assert(row && 'userId' in row && 'fileId' in row, 'userId is required')
 		if (!this.userIsActive(row.userId)) return
 		if (event.command === 'insert') {
-			this.sqlite.exec(
-				`INSERT INTO user_file_subscriptions (userId, fileId) VALUES (?, ?) ON CONFLICT (userId, fileId) DO NOTHING`,
-				row.userId,
-				row.fileId
-			)
-			assert(typeof row.isFileOwner === 'boolean', 'isFileOwner is required')
 			if (!row.isFileOwner) {
-				// need to dispatch a file creation event to the user
-				this.getFileRecord(row.fileId).then((file) => {
-					// check that the subscription wasn't deleted before we managed to fetch the file record
-					const sub = this.sqlite
-						.exec(
-							`SELECT * FROM user_file_subscriptions WHERE userId = ? AND fileId = ?`,
-							row.userId,
-							row.fileId
-						)
-						.toArray()[0]
-					if (!sub) return
-
-					// alright we're good to go
-					this.messageUser(row.userId, {
-						type: 'row_update',
-						row: file as any,
-						table: 'file',
-						event: 'insert',
-						userId: row.userId,
-					})
-				})
```

Comment:
it is no longer practical to send random updates to the user like this, so I made it the responsibility of the UserDataSyncer to fetch and delete the file records for guest files on an ad-hoc basis.